lib/puppet/defaults.rb

```diff
@@ -1692,11 +1692,17 @@ def self.default_diffargs
     :archive_files => {
         :type     => :boolean,
         :default  => false,
-        :desc     => "During an inspect run, whether to archive files whose contents are audited to a file bucket.",
+        :desc     => "During an inspect run, whether to archive files whose contents are audited to a file bucket. Note that the `inspect` command is deprecated.",
+        :hook => proc { |value|
+          Puppet.deprecation_warning(_("Setting 'archive_files' is deprecated. It will be removed in a future release along with the `inspect` command."))
```

Comment:
Good question @Magisus. We're not consistent in how we issue deprecation warnings for settings. Sometimes it's whenever the value is set, e.g. `trusted_server_facts`, or other times only in strict mode, e.g. `data_binding_terminus`.

Moving forward I think we adopt the latter approach. Don't spam users, but give them an opportunity to run with strict.

In this particular case, these settings are in the `:inspect` section, but nothing in puppet will `Puppet.settings.use(:inspect)` them. So they never end up in a settings catalog, like can happen with file/dir settings (which can cause deprecation warnings each time puppet runs).

AFAIK these settings only exist so that the `inspect` application can have default values, but override them on the command line. If you have these settings in `puppet.conf` and you run puppet (apply, agent, etc), does it issue a deprecation? If so, then I would make it only do that in `strict` mode.